dallinger/command_line/docker_ssh.py

```diff
@@ -363,6 +391,13 @@ def deploy(
     dashboard_user = config.get("dashboard_user", "admin")
     dashboard_password = config.get("dashboard_password", secrets.token_urlsafe(8))
 
+    recruiter = by_name(config["recruiter"])
+    if not recruiter.validate_config(mode_from_command=mode):
+        print(
+            "When deploying to MTurk either `open_recruitment` must be `True` in the config or the `--open-recruitment` flag must be provided in the deploy command."
+        )
+        raise click.Abort
```

Comment:
@fmhoeger can you think a bit more carefully about this? Why is MTurk logic located here instead of within `validate_config`? 

Instead of testing for a bool returned from `validate_config`, I think you should be raising the error within `validate_config` itself. That will help you to keep separate recruiter logic separate, and it will help readability because you will construct the error message next to the place where the condition was tested.

Reply:
My rational of thinking was that we cannot be sure if `raise click.Abort` should be called as `validate_config` could have been called from a context other than command line, one where `raise click.Abort` makes no sense.

Reply:
I think it's best therefore to raise a ValueError and don't worry about click.Abort. If you want you could catch and reraise that ValueError via click.Abort, but I think it is unnecessary.

Reply:
I see. Reworked the `validate_config` logic.